Source/JavaScriptCore/runtime/ISO8601.cpp

```diff
@@ -843,7 +905,37 @@ static std::optional<CalendarRecord> parseCalendar(StringParsingBuffer<Character
     if (*buffer != ']')
         return std::nullopt;
     buffer.advance();
-    return CalendarRecord { WTFMove(result) };
+
+    return CalendarRecord { isCritical, false, WTFMove(result) };
+}
+
+template<typename CharacterType>
+static std::tuple<std::optional<CalendarRecord>, std::optional<CalendarRecord>>
+parseCalendar(StringParsingBuffer<CharacterType>& buffer)
+{
+// https://tc39.es/proposal-temporal/#prod-Annotations
+//  Annotations :::
+//      Annotation Annotationsopt
+
+    std::optional<CalendarRecord> first = parseOneCalendar(buffer);
+    std::optional<CalendarRecord> second;
+    if (first && canBeCalendar(buffer))
+        second = parseOneCalendar(buffer);
+    // Check for multiple calendars and critical flag
+    // https://tc39.es/proposal-temporal/#sec-temporal-parseisodatetime
+    // step 4(a)(ii)(2)(c)(ii)
+    if (first && second) {
+        if (first->m_critical || second->m_critical)
+            first = second = std::nullopt;
+    }
+    // Check for unknown annotations with critical flag
+    // https://tc39.es/proposal-temporal/#sec-temporal-parseisodatetime
+    // step 4(a)(ii)(2)(d)(i)
+    if ((first && first->m_critical && first->m_unknown)
+        || (second && second->m_critical && second->m_unknown)) {
+        first = second = std::nullopt;
```

Comment:
WebKit coding style doesn't use this kind of assignment.
We should return std::nullopt here.

Reply:
Fixed in [9ac33a5](https://github.com/WebKit/WebKit/pull/36361/commits/9ac33a55dfdaf9198aace8a87462d875cf01e0d5)